docs/Navigation.md

```diff
@@ -77,13 +77,24 @@ Parameters:
 
   * `<lon>` - Longitude.
 
-  * `<alt>` - Altitude in cm.
+  * `<alt>` - Altitude in cm. See `p3` bit 0 for datum definition.
 
   * `<p1>` - For a RTH waypoint, p1 > 0 enables landing. For a normal waypoint it is the speed to this waypoint (cm/s), it is taken into account only for multicopters and when > 50 and < nav_auto_speed. For POSHOLD TIME waypoint it is time to loiter in seconds. For JUMP it is the target WP **index** (not number). For SET_HEAD, it is the desired heading (0-359) or -1 to cancel a previous SET_HEAD or SET_POI.
 
   * `<p2>` - For a POSHOLD TIME it is the speed to this waypoint (cm/s), it is taken into account only for multicopters and when > 50 and < nav_auto_speed. For JUMP it is the number of iterations of the JUMP.
 
-  * `<p3>` - Reserved for future use. If `p2` is provided, then `p3` is also required.
+  * `<p3>` - A  bitfield with four bits reserved for user specified actions. It is anticipated that these actions will be exposed through the logic conditions.
+      * Bit 0 - Altitude (`alt`) : Relative (to home altitude) (0) or Absolute (AMSL) (1).
+	  * Bit 1 - WP Action 1
+	  * Bit 2 - WP Action 2
+      * Bit 3 - WP Action 3
+      * Bit 4 - WP Action 4
+	  * Bits 5 - 15 : undefined / reserved.
+
+      Note:
+
+	  * If `p2` is specified, then `p3` is also required.
+	  * `p3` is only defined for geographic WP types (WAYPOINT, POSHOLD_TIME, SET_POI, LAND). The affect of specifying a non-zero `p3` for other WP types is undefined.
```

Comment:
Don't think SET_POI is applicable here ?

Reply:
Well it's a geographical WP in that it has a "position". 
But I agree, as the machine never goes to it, setting actions is not really appropriate.
I'll remove it for the document.